### PR TITLE
add initial tox config for certification and partner testing

### PR DESCRIPTION
**Pull request by thedoubl3j**
This PR contains the initial pass of linters/testing to be included in the certification pipeline for relevant python content for EDA specifically. 

A version of this file will live inside galaxy importer that will need be updated if this file ever changes (PE will manage that)

Feedback around what to use, why something was selected is welcomed. 

For local use and testing, place the tox.ini file in the top level of the collection and then call simply call `tox` from the CLI in that directory. 

There is a known bug currently with pylint around one or more of the directories not existing, we are looking at ways to get around that inside of the tox file. Help wanted if folks have ideas. *solved*

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ruff
+    darglint
+    pylint
+    black
+
+[testenv:format-black]
+deps =
+    black
+commands =
+    black -l 100 event-driven-ansible/extensions/eda/plugins
```
Comment (alisonlhart) [inline tox.ini:13]: The path needs to be updated to `/extensions/eda/plugins` for all the linters. The path included the collection name only when it was outside of the root dir.

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+deps = ruff
+commands = ruff check --select ALL --ignore INP001 -q /extensions/eda/plugins
+
+
+[testenv:darglint]
+deps = darglint
+commands = darglint -s sphinx -z full /extensions/eda/plugins
+
+# depending on what kind of plugins you have, remove the line you don't need (i.e event_sources or event_filters, remove the pylint call for the other one)
+# depending on how your collection and repo is structured, you may need to change the path to each type of plugin
```
Comment (alisonlhart) [inline tox.ini:18]:
```suggestion
# depending on how your collection and repo is structured, you may need to change the path to each type of plugin
# if pylint warns about missing __init__.py files in directories, there's no need to include them if you ensure that the paths in the below pylint `commands` point directly to the *.py files under event_source/ and event_filters, as shown in the template path here
```

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    pylint
+
+[testenv:ruff]
+deps = ruff
+commands = ruff check --select ALL --ignore INP001 -q /extensions/eda/plugins
+
+
+[testenv:darglint]
+deps = darglint
+commands = darglint -s sphinx -z full /extensions/eda/plugins
```
Comment (alisonlhart) [inline tox.ini:15]:
```suggestion
commands = darglint -s numpy -z full /extensions/eda/plugins
```

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[tox]
```
Comment (alisonlhart) [inline tox.ini:1]:
```suggestion
# this template can be adjusted as needed for your CI environment. the paths in the `commands` of each tox env linter can be changed if the tox file sits in a different location than expected in this tempate
[tox]
```

**Review by alisonlhart — DISMISSED** [review]
LGTM :+1:

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+[testenv:darglint]
+deps = darglint
+commands = darglint -s numpy -z full /extensions/eda/plugins
+
+# depending on what kind of plugins you have, remove the line you don't need (i.e event_sources or event_filters, remove the pylint call for the other one)
+# depending on how your collection and repo is structured, you may need to change the path to each type of plugin
+# if pylint warns about missing __init__.py files in directories, there's no need to include them if you ensure that the paths in the below pylint `commands` point directly to the *.py files under the event_source/ and event_filter/ dirs, as shown in the template path here
+[testenv:pylint-event-source]
+deps = pylint
+commands =
```
Comment (alisonlhart) [inline tox.ini:23]:
```suggestion
commands = pylint /extensions/eda/plugins/event_source/*.py --output-format=parseable -sn --disable R0801
```

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+deps = darglint
+commands = darglint -s numpy -z full /extensions/eda/plugins
+
+# depending on what kind of plugins you have, remove the line you don't need (i.e event_sources or event_filters, remove the pylint call for the other one)
+# depending on how your collection and repo is structured, you may need to change the path to each type of plugin
+# if pylint warns about missing __init__.py files in directories, there's no need to include them if you ensure that the paths in the below pylint `commands` point directly to the *.py files under the event_source/ and event_filter/ dirs, as shown in the template path here
+[testenv:pylint-event-source]
+deps = pylint
+commands = 
+    pylint /extensions/eda/plugins/event_source/*.py --output-format=parseable -sn --disable R0801
```
Comment (alisonlhart) [inline tox.ini:24]:
```suggestion
```

File: tox.ini [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+[testenv:pylint-event-filter]
+deps = pylint
+commands =
```
Comment (alisonlhart) [inline tox.ini:28]:
```suggestion
commands = pylint /extensions/eda/plugins/event_filter/*.py --output-format=parseable -sn --disable R0801
```

File: tox.ini [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+# if pylint warns about missing __init__.py files in directories, there's no need to include them if you ensure that the paths in the below pylint `commands` point directly to the *.py files under the event_source/ and event_filter/ dirs, as shown in the template path here
+[testenv:pylint-event-source]
+deps = pylint
+commands = 
+    pylint /extensions/eda/plugins/event_source/*.py --output-format=parseable -sn --disable R0801
+
+[testenv:pylint-event-filter]
+deps = pylint
+commands = 
+    pylint /extensions/eda/plugins/event_filter/*.py --output-format=parseable -sn --disable R0801
```
Comment (alisonlhart) [inline tox.ini:29]:
```suggestion
```

File: tox.ini (hunk 7 above)
Comment (alisonlhart) [reply tox.ini:28]: Placing it on the same line as command. Was causing some issues when I tested it with it on a line below.

File: tox.ini [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# this template can be adjusted as needed for your CI environment. the paths in the `commands` of each tox env linter can be changed if the tox file sits in a different location than expected in this template
+[tox]
+envlist = ruff, darglint, pylint
```
Comment (alisonlhart) [inline tox.ini:3]:
```suggestion
envlist = ruff, darglint, pylint-event-source, pylint-event-filter
```

**Review by benthomasson — APPROVED** [review]

**Comment by thedoubl3j** [comment]
I will squash all of these or if merger wants to, go for it once it is approved

**Review by alisonlhart — APPROVED** [review]

**Review by benthomasson — APPROVED** [review]

**Review by alisonlhart — APPROVED** [review]

**Review by ddonahue007 — APPROVED** [review]
lgtm

**Pull request merged**